### PR TITLE
[HttpFoundation] added functionality to HeaderBag::contains()

### PR DESCRIPTION
**Pull request by rupertjeff**
[HttpFoundation] added functionality to `HeaderBag::contains()` to do a string comparison if the initial array comparison fails

This change was made to clear up an ambiguity in the documentation for `HeaderBag::contains()`. Read the fixed ticket mentioned below for more info.

More than happy to make any changes needed! (I'm willing to make an update to the documentation to make it more explicit, if that is more agreeable.)

| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #16646 |
| License | MIT |
| Doc PR | n/a |

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/Command/ServerStartCommand.php [hunk 1]
```diff
@@ -74,7 +74,7 @@ protected function configure()
      */
     protected function execute(InputInterface $input, OutputInterface $output)
     {
-        $output = new SymfonyStyle($input, $output);
+        $output = new SymfonyStyle($input, $cliOutput = $output);
```
Comment (derrabus) [inline src/Symfony/Bundle/FrameworkBundle/Command/ServerStartCommand.php]: What's the purpose of this change?

**Comment by derrabus** [comment]
:-1: Until now, the method you've changed reported `true` on exact matches of the header's content only. This change will break code that relies on the current behavior.

Apart from that, a commit slipped into your PR that seems unrelated to the change you're proposing. You might want to rebase your commit against the current master.

**Comment by rupertjeff** [comment]
Weird. I rebased with master and that commit appeared.

So, the better option here is to update the documentation, even though the code itself is easily misconstrued, correct? I'm happy to remove this PR and submit one to the documentation instead.

Thoughts?

**Comment by derrabus** [comment]
The semantics of `HeaderBag::contains($key, $value)` is: Does this bag of headers contain a header named `$key` with the exact value `$value`? There's nothing wrong with these semantics and the code does reflect them clearly, imho. The only problem I see is that the documentation of that method leaves room for misinterpretations.

Since I'm not a maintainer of this project, I cannot reject your pull request. But I would not recommend to merge it because it changes the behavior of a public method. And I fear that this change will have an impact on existing code that we can hardly estimate.

Clarifying the documentation would be the way to go, imho.

**Comment by stof** [comment]
I agree with @derrabus here. This change is a BC break and so cannot be accepted.

If this is really the behavior you need for your use case, you can totally run it without using ``HeaderBag::contains``.

Feel free to send a PR updating the phpdoc to make it easier to understand what the method does.
